django/db/models/query.py

```diff
@@ -1045,7 +1045,7 @@ def annotate(self, *args, **kwargs):
                 clone.query.add_annotation(annotation, alias, is_summary=False)
 
         for alias, annotation in clone.query.annotations.items():
-            if alias in annotations and annotation.contains_aggregate:
+            if alias in annotations and getattr(annotation, 'contains_aggregate', False):
```

Comment:
Out of curiosity, what's `type(annotation)` that crashes here? I thought this would be `CombinedExpression` from the test below.

Reply:
`ResolvedOuterRef`, added test will crash in some other place without your great work on `Subquery` :+1:

Reply:
That's why I waited few months with sending PR :smile:

Reply:
Any thoughts about adding `ResolvedOuterRef.contains_aggregate = False` instead?

`F` resolves to `Col` which inherits `Expression.contains_aggregate` and `OuterRef` resolves to `ResolvedOuterRef` which doesn't have `contains_aggregate`. It feels like requiring resolved expressions to have a `contains_aggregate` property is a better interface than using `getattr` for this single case here.

Reply:
Agreed :+1: Let's revert to the 1st version of this patch.